components/Tabs/Tabs.js

```diff
@@ -1,72 +1,85 @@
+    this.tabElement = tabElement;
     
     // Get the `data-tab` value from this.tabElement and store it here
-    // this.tabData = ; 
+    this.tabData = this.tabElement.dataset.tab;
+    console.log(this.tabData)
+    // We need to find out if a user clicked 'all' cards or a specific category.
+    // Follow the instructions below to accomplish this task:    
     
-    // We need to find out if a user clicked 'all' cards or a specific category.  Follow the instructions below to accomplish this task:    
-    
-    /* <- Delete this comment block when you work on the if statement
+    // Delete this comment block when you work on the if statement
     // Check to see if this.tabData is equal to 'all'
-    if(){
+    if(this.tabData === 'all'){
       // If `all` is true, select all cards regardless of their data attribute values
-      // this.cards = ;
+    this.cards = document.querySelectorAll('.card')
     } else {
       // else if `all` is false, only select the cards with matching this.tabData values
-      // this.cards = ;
-    }
-    /* <- Delete this comment block when you work on the if statement
+    this.cards = document.querySelectorAll(`.card[data-tab='${this.tabData}']`);
+   console.log(this.cards)
+  };
+    // Delete this comment block when you work on the if statement
+// console.log(this.cards);
+     // Map over the newly converted NodeList we just created in our if statement above.
+    //  Convert each this.cards element into a new instance of the TabCard class.
+    // Pass in a card object to the TabCard class. 
+    this.cards = Array.from(this.cards).map(element => new TabCard(element));
 
-     // Map over the newly converted NodeList we just created in our if statement above. Convert each this.cards element into a new instance of the TabCard class. Pass in a card object to the TabCard class. 
-    // this.cards = Array.from(this.cards).map();
+    // this.tabCard = new TabCard(this.cards[0]);
 
     // Add a click event that invokes this.selectTab
-    // this.tabElement.addEventListener();
-  }
+    this.tabElement.addEventListener('click', () => this.selectTab());
+  };
 
   selectTab(){
 
     // Select all elements with the .tab class on them
-    // const tabs = document.querySelectorAll();
+    const tabs = document.querySelectorAll('.tab');
     
     // Iterate through the NodeList removing the .active-tab class from each element
-    // tabs.forEach()
+    tabs.forEach(tabList => tabList.classList.remove('.active-tab'));
 
     // Select all of the elements with the .card class on them
-    // const cards = ;
-
+    const cards = document.querySelectorAll('.card');
+    // console.log(cards)
     // Iterate through the NodeList setting the display style each one to 'none'
-    // cards.forEach()
+    cards.forEach(tabList => tabList.style.display = "none");
     
     // Add a class of ".active-tab" to this.tabElement
-    // this.tabElement;
+    this.tabElement.classList.add('.active-tab');
```

Comment:
remove the dot before active-tab.  because classList recognizes you're working with classes, you just have to pass in the name of the class in quotes.